doc/glossary.rst

```diff
@@ -1336,7 +1341,13 @@ Methods
         often the likelihood of the data under the model.
 
     ``score_samples``
-        TODO
+        A method that returns the likelihood of given samples.
+
+        For density estimation, it returns the value (log) of the density of the
+        samples.
+
+        For outlier detection, it returns a score for the sample based on its
+        likelihood (thus if it is an outlier, it's not likely at all).
```

Comment:
I think likelihood is more well-defined than what we have as the output of`score_samples` in different places. Maybe something like:
```
A method that returns a score for each given sample. The exact definition
of *score* varies from one class to another. 

In the case of density estimation, it can be the log density model on the data,
and in the case of outlier detection, it can be the opposite of the outlier factor
of the data.
```